src/Microsoft.Health.Fhir.Core/Configs/OperationsConfiguration.cs

```diff
@@ -22,5 +24,27 @@ public class OperationsConfiguration
         public IntegrationDataStoreConfiguration IntegrationDataStore { get; set; } = new IntegrationDataStoreConfiguration();
 
         public ImportTaskConfiguration Import { get; set; } = new ImportTaskConfiguration();
+
+        /// <summary>
+        /// Removes queues based on the enabled status of the operations.
+        /// </summary>
+        public void RemoveDisabledQueues()
```

Comment:
Maybe a silly question, why do we add them then remove instead of skipping to add?

Reply:
I thought the same thing initially. 😆 The main reasoning is this approach allows default configuration applied in `appsettings.json` which is parsed to the configuration objects. I think this is the correct approach vs applying overrides via environment variables/appsettings overrides in the hosting environment which have to be maintained separately.

So once we parse the defaults into the configuration class, why have the removal logic here? I'm thinking keeping it in this class encapsulates the logic here vs polluting the registration classes with operation specific logic. It just exposes a hook to be called.

I was on the fence about implementation on this - happy to change the approach if there are thought a different direction.

Reply:
@brendankowitz - I could use a custom `ConfigurationProvider` instead like below. Thoughts?? I think this would intercept the configuration of the class.

https://github.com/microsoft/fhir-server/blob/cb53a355a88c50456c4542484a96512642b54b1a/src/Microsoft.Health.Fhir.Api/Features/Binders/DictionaryExpansionConfigurationProvider.cs#L22

Reply:
In thinking about it, perhaps this is calling more to consolidate these settings into one place. Right now we enable/disable in one place, configure job settings in another and remove as a post step. It might make more sense to move all these into a single config which would make this more discoverable / maintainable